lib/ansible/playbook/play_context.py

```diff
@@ -424,7 +423,7 @@ def detect_su_prompt(data):
                     return bool(SU_PROMPT_LOCALIZATIONS_RE.match(data))
 
                 prompt = detect_su_prompt
-                becomecmd = '%s %s %s -c "%s -c %s"' % (exe, flags, self.become_user, executable, success_cmd)
+                becomecmd = '%s %s %s -c %s' % (exe, flags, self.become_user, pipes.quote("%s -c %s" % (shell, success_cmd)))
```

Comment:
This may be unnecessary and obvious, but given that this patch still hasn't been merged, I wanted to point out that the change shown above from ``"%s -c %s"`` → ``%s`` doesn't mean that we stop calling ``sh -c %s`` (which is something I've discussed in the past). The entire shell+command invocation is quoted as one unit (which isn't clear without scrolling all the way to the right).